### PR TITLE
[FCE-890] Update documentation to use new `useConnection` hook

### PR DESCRIPTION
**Pull request by maksg**
## Description

Changed `usePeerStatus` and `useReconnection` to `useConnection`.

### CODE REVIEW

**Review by MiloszFilimowski — APPROVED** [review]
👍

**Comment by mironiasty** [comment]
I think we should also update `joinRoom`/`leaveRoom`  docs, to show that these methods are part of `useConnection` hook.

**Comment by maksg** [comment]
@mironiasty @MiloszFilimowski I have updated the docs for `joinRoom` and `leaveRoom`

File: docs/react-native/connecting.mdx [hunk 1]
```diff
@@ -46,7 +46,8 @@ follow our [server setup instructions](/production/server).
 In order to connect, you just need to call `joinRoom` method with data from previous step:
 
 ```ts
-import { joinRoom } from "@fishjam-cloud/react-native-client";
+import { useConnection } from "@fishjam-cloud/react-native-client";
+const { joinRoom } = useConnection();
```
Comment (MiloszFilimowski) [inline docs/react-native/connecting.mdx:50]: **nitpick:** Can you add a new line after the import? 🙏

File: docs/react-native/connecting.mdx [hunk 2]
```diff
@@ -56,7 +57,8 @@ await joinRoom(fishjamUrl, peerToken);
 In order to close connection, you have to call `leaveRoom` method.
 
 ```ts
-import { leaveRoom } from "@fishjam-cloud/react-native-client";
+import { useConnection } from "@fishjam-cloud/react-native-client";
+const { leaveRoom } = useConnection();
```
Comment (MiloszFilimowski) [inline docs/react-native/connecting.mdx:61]: **nitpick:** Same request 🙏

**Review by MiloszFilimowski — APPROVED** [review]

**Comment by maksg** [comment]
@MiloszFilimowski Done!

**Review by mironiasty — APPROVED** [review]

**Pull request merged**